File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/ahn-product-1-0/ConstPayoff.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ConstPayoff : ISerialized
    {
        public ConstPayoff(XmlNode xmlNode)
        {
            XmlNode constValueNode = xmlNode.SelectSingleNode("constValue");
            
            if (constValueNode != null)
            {
                if (constValueNode.Attributes["href"] != null || constValueNode.Attributes["id"] != null) 
                {
                    if (constValueNode.Attributes["id"] != null) 
                    {
                        constValueIDRef_ = constValueNode.Attributes["id"].Value;
                        XsdTypeDouble ob = new XsdTypeDouble(constValueNode);
                        IDManager.SetID(constValueIDRef_, ob);
                    }
                    else if (constValueNode.Attributes["href"] != null)
                    {
                        constValueIDRef_ = constValueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        constValue_ = new XsdTypeDouble(constValueNode);
                    }
                }
                else
                {
                    constValue_ = new XsdTypeDouble(constValueNode);
                }
            }
            
        
        }
        
    
        #region ConstValue_
        private XsdTypeDouble constValue_;
        public XsdTypeDouble ConstValue_
        {
            get
            {
                if (this.constValue_ != null)
                {
                    return this.constValue_; 
                }
                else if (this.constValueIDRef_ != null)
                {
                    constValue_ = IDManager.getID(constValueIDRef_) as XsdTypeDouble;
                    return this.constValue_; 
                }
                else
                {
                      return this.constValue_; 
                }
            }
            set
            {
                if (this.constValue_ != value)
                {
                    this.constValue_ = value;
                }
            }
        }
        #endregion
        
        public string constValueIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/ahn-product-1-0/ConstPayoff.cpp
// ConstPayoff.cpp 
#include "ConstPayoff.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ConstPayoff::ConstPayoff(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //constValueNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* constValueNode = xmlNode->FirstChildElement("constValue");

   if(constValueNode){constValueIsNull_ = false;}
   else{constValueIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- constValueNode , address : " << constValueNode << std::endl;
   #endif
   if(constValueNode)
   {
      if(constValueNode->Attribute("href") || constValueNode->Attribute("id"))
      {
          if(constValueNode->Attribute("id"))
          {
             constValueIDRef_ = constValueNode->Attribute("id");
             constValue_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(constValueNode));
             constValue_->setID(constValueIDRef_);
             IDManager::instance().setID(constValueIDRef_,constValue_);
          }
          else if(constValueNode->Attribute("href")) { constValueIDRef_ = constValueNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { constValue_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(constValueNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDouble> ConstPayoff::getConstValue()
{
   if(!this->constValueIsNull_){
        if(constValueIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDouble>(IDManager::instance().getID(constValueIDRef_));
        }else{
             return this->constValue_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDouble>();
   }
}
}

